### PR TITLE
[INF] Enable torch compile for inference

### PR DESCRIPTION
**Pull request by oelayan7**

### CODE REVIEW

**Comment by delock** [comment]
Hi @oelayan7 do you have sample code using this feature?  Or does it turn on by default?  I wonder if I can try it with CPU accelerator.

**Comment by oelayan7** [comment]
Hi @delock, 
No, it isn't run by default. The function .compile() needs to be called. 
This changed was adopted in a similar way that was done in zero and for training, which used CompileWrapper that was removed in https://github.com/microsoft/DeepSpeed/pull/5581 and a new approach was introduced.
This change was adopted from the all old PRs before.

**Comment by oelayan7** [comment]
@tjruwase the failing tests are because an HTTP failure with HF. 
Can you please retrigger? (couldn't find a way to do that)

**Comment by oelayan7** [comment]
@tjruwase @tohtana @umchand Can anyone trigger the CI please?

**Review by tohtana — APPROVED** [review]
Thank you @oelayan7 !

File: deepspeed/inference/engine.py [hunk 1]
```diff
@@ -6,13 +6,15 @@
 import torch
 import time
 import os
+import deepspeed.utils.nvtx
```
Comment (loadams) [inline deepspeed/inference/engine.py:9]: @oelayan7 - looks like this is imported but unused, and that's what is causing the formatting failures, could you check and remove this line if so?

Comment (loadams) [reply deepspeed/inference/engine.py:9]: Seems we may need this merged first?

#5697

Comment (tohtana) [reply deepspeed/inference/engine.py:9]: Yes, you are right. Sorry for the confusion

**Review by umchand — APPROVED** [review]

File: deepspeed/inference/engine.py (hunk 1 above)
Comment (loadams) [reply deepspeed/inference/engine.py:9]: That PR is merged now.

**Review by loadams — APPROVED** [review]

**Comment by oelayan7** [comment]
@loadams Thanks! 
I fixed the formatting issue, can you please retrigger?

**Comment by oelayan7** [comment]
@loadams I don't think the failures are related.

**Comment by loadams** [comment]
> @loadams I don't think the failures are related.

Correct, working on getting this merged despite some hiccups in our tests not related to your PR.

**Comment by loadams** [comment]
> > @loadams I don't think the failures are related.
> 
> @oelayan7 Correct, working on getting this merged despite some hiccups in our tests not related to your PR.

@oelayan7 - we believe we should be merging all the CI fixes today and will get this merged shortly after that.

**Pull request merged**